dask/array/core.py

```diff
@@ -3010,37 +3014,80 @@ def normalize_chunks(chunks, shape=None, limit=None, dtype=None, previous_chunks
 
     >>> normalize_chunks((), shape=(0, 0))
     ((0,), (0,))
+
+    Use ``numblocks`` to replicate a chunk shape a given number of times.
+    Note that ``'auto'``, -1, and None cannot be used if only ``numblocks`` is given.
+
+    >>> normalize_chunks(4, numblocks=(2, 3))
+    ((4, 4), (4, 4, 4))
+
+    Use ``shape`` and ``numblocks`` without any chunksize
+
+    >>> normalize_chunks(None, shape=(10, 5), numblocks=(2, 2))
+    ((5, 5), (3, 2))
     """
+    ndim: int | None = None
+    if shape is not None:
+        ndim = len(shape)
+    if numblocks is not None:
+        if ndim is not None:
+            if len(numblocks) != ndim:
+                raise ValueError(
+                    f"`shape` indicates {ndim} dimension(s), but `numblocks` indicates {len(numblocks)}."
+                )
+        ndim = len(numblocks)
```

Comment:
I'm wondering if this whole bit could be separated out into a separate function maybe called `_get_ndim` or something.